integ-tests/hadoop/src/test/java/com/google/integration/SchemaTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.integration;
+
+import static com.google.base.Constants.PASSWORD_DB;
+import static com.google.base.Constants.SQL_TEST_DATA_BASE_PATH;
+import static com.google.base.Constants.URL_DB;
+import static com.google.base.Constants.USERNAME_DB;
+import static com.google.sql.SqlHelper.getSql;
+import static java.lang.String.format;
+
+import com.google.base.TestBase;
+import com.google.common.collect.ImmutableSet;
+import com.google.dumper.DumperUtils;
+import com.google.proto.HiveSchema.Schema;
+import com.google.sql.SqlHelper;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SchemaTest extends TestBase {
+
+  private static Connection connection;
+  private static final Logger LOGGER = LoggerFactory.getLogger(SchemaTest.class);
+  private static final String GET_TABLES_QUERY = SQL_TEST_DATA_BASE_PATH + "get_tables.sql";
+  private static final String GET_SCHEME_QUERY = SQL_TEST_DATA_BASE_PATH + "get_table_schema.sql";
+  private static final String SQL_ERROR_MESSAGE = "Cannot execute query %n%s%n";
+
+  @BeforeClass
+  public static void beforeClass() throws SQLException {
+    connection = DriverManager.getConnection(URL_DB, USERNAME_DB, PASSWORD_DB);
+  }
+
+  @Test
+  public void schemaTest() throws IOException, SQLException {
+    ImmutableSet<Schema> dumperSet = DumperUtils.readSchemas();
+
+    Set<String> dbList = new HashSet<>(Arrays.asList(System.getenv("HADOOP_SCHEMAS").split(",")));
+    Map<String, Set<String>> dbTables = getTables(dbList);
+    ImmutableSet<Schema> hiveSet = getSchemas(dbTables);
+
+    assertSetsAreEqual(hiveSet, dumperSet);
+  }
+
+  private ImmutableSet<Schema> getSchemas(Map<String, Set<String>> dbTables) {
+    List<Schema> hiveList = new ArrayList<>();
+    dbTables
```

Comment:
Let's convert it to regular For-Each, please.

Reply:
Fixed, please check